### PR TITLE
Adding a 'Create a new post' feature

### LINKED ISSUE #96: Create a UI for adding a new post

### PR DESCRIPTION
**Pull request by Shoval-Ovadia**
A registered user can add a new post at: '127.0.0.1:8000/createPost/'.

- created a view function - Post/views.py modified.
- added a new URL for this feature -  picATrip/urls.py modified.
- created a form for this feature - Post/forms.py created.
- created a HTML page -  Post/templates/Post/createPost.html created.

These changes were made to create a user interface for the feature of creating a new post.

closes #96

### CODE REVIEW

File: Post/forms.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from django import forms
+from .models import Post
+
+
+class AddAPostForm(forms.ModelForm):
```
Comment (maayanhd) [inline Post/forms.py:5]: maybe AddaPostForm will break the capital letters sequence, but that's only one opinion, to just AddPostForm

File: Post/views.py [hunk 2]
```diff
@@ -11,8 +12,20 @@ class PostListView(ListView):
     context_object_name = 'posts'
 
 
+@login_required
+def AddAPost(request):
+    form = AddAPostForm(request.POST)
+    if form.is_valid():
+        form.save()
+        return redirect("http://127.0.0.1:8000/postList")
```
Comment (Yarboa) [inline Post/views.py:20]: redirect("http://127.0.0.1:8000/postList")
Will not work in production, some think like that
return HttpResponseRedirect('postList?')

File: Post/views.py [hunk 3]
```diff
@@ -11,8 +12,20 @@ class PostListView(ListView):
     context_object_name = 'posts'
 
 
+@login_required
+def AddAPost(request):
```
Comment (maayanhd) [inline Post/views.py:16]: maybe AddPost is a bit more clear

**Review by Yarboa — COMMENTED** [review]
Nice PR, did you think how to test it ? 
It is advised to Propose new feature with tests, to test it, at least a minimum

**Review by maayanhd — COMMENTED** [review]
very nice, in later version maybe we can add image upload and find a way to generate a URL for the uploaded image in order to save it in the DB but that's for later. Just small naming changed that are debatable.

File: Post/views.py (hunk 2 above)
Comment (Shoval-Ovadia) [reply Post/views.py]: I fixed it.

File: Post/views.py (hunk 3 above)
Comment (Shoval-Ovadia) [reply Post/views.py]: I changed it to CreateNewPost, is it ok? @maayanhd

File: Post/forms.py (hunk 1 above)
Comment (Shoval-Ovadia) [reply Post/forms.py]: I changed it to CreatePostForm. Is it better? @maayanhd

**Comment by Shoval-Ovadia** [comment]
> Nice PR, did you think how to test it ?
> It is advised to Propose new feature with tests, to test it, at least a minimum

**Comment by Shoval-Ovadia** [comment]
> Nice PR, did you think how to test it ?
> It is advised to Propose new feature with tests, to test it, at least a minimum

I will try to add tests today.

**Review by nadavspinco — APPROVED** [review]

File: Post/templates/Post/createPost.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{% extends "pickATrip_django_apps/base.html" %}
+{% block content %}
+{% load post_extra %}
```
Comment (Leeadar) [inline Post/templates/Post/createPost.html:3]: What is post_extra ?

File: Post/views.py [hunk 5]
```diff
@@ -11,8 +12,20 @@ class PostListView(ListView):
     context_object_name = 'posts'
 
 
+@login_required
+def CreateNewPost(request):
+    form = CreatePostForm(request.POST)
+    if form.is_valid():
+        form.save()
```
Comment (Leeadar) [inline Post/views.py:19]: Maybe add here a success messages that the post has created. (Inside the if block)
You can do so by adding `messages.success(request, 'The post has been created' )`

Comment (Shoval-Ovadia) [reply Post/views.py:20]: @Leeadar, Nice! 
I added the message.

File: Post/templates/Post/createPost.html (hunk 4 above)
Comment (Shoval-Ovadia) [reply Post/templates/Post/createPost.html]: It wasn't necessary, I deleted it.

**Review by Leeadar — COMMENTED** [review]
Can you add the links to the new pages into the navigation bar?
I also think that we need to delete the name of the poster from the form, and insert instead the username ( and not let the user to insert a random name).

**Comment by Shoval-Ovadia** [comment]
> Can you add the links to the new pages into the navigation bar?
> I also think that we need to delete the name of the poster from the form, and insert instead the username ( and not let the user to insert a random name).

Hi @Leeadar,
I think we should display the post list on the main page. 
If we will do that, it won't be necessary to add it to the navigation bar. 

I will change the field from 'nameOfPoster' to the user. 
I think that after that, the user will not have to enter his username. what do you think?

**Comment by Leeadar** [comment]
So in the meantime you can add the link in the navigation bar, or change the homepage to be the post_list page.
Also we need to add a link to the create a new post, you can add that link inside the base.html (inside the is authenticated block).

`I think that after that, the user will not have to enter his username. what do you think?`
   Yes, in my opinion the user shouldn't enter a username at all

**Review by Yarboa — APPROVED** [review]
LGTMT, +2, lets wait another CR

**Review by maayanhd — APPROVED** [review]

**Comment by Shoval-Ovadia** [comment]
> So in the meantime you can add the link in the navigation bar, or change the homepage to be the post_list page.
> Also we need to add a link to the create a new post, you can add that link inside the base.html (inside the is authenticated block).
> 
> `I think that after that, the user will not have to enter his username. what do you think?`
> Yes, in my opinion the user shouldn't enter a username at all

I added the pages to the navigation bar.

**Review by Leeadar — APPROVED** [review]
LGTM

**Comment by Shoval-Ovadia** [comment]
The last change I did is on the fields of the form (I renamed the fields).

**Review by nadavspinco — APPROVED** [review]

**Pull request merged**